### ISSUE TITLE
AWS Batch job description additional mounts

### ISSUE THREAD

**Issue by microbioticajon** [broadinstitute/cromwell#6334]
Hi Guys,

This is more of a question/request than a bug report. Apologies if this is not the place to ask.

Im trying to run Cromwell with an AWS backend. A number of our workflows make extensive use of very large reference files. To avoid localising the same huge file over and over (wasting time and space) I want to copy these reference files to an additional volume during batch node initialisation and mount to each container (rather than using File arguments I would use a simple String argument to prevent localisation - I appreciate this is a hack). I am already doing this with a different pipeline framework with some success, however it requires the JobDefinition to specify the mount locations between the node(host) and job container

Is it possible to provide additional mount/volume instructions to the aws batch backend in the cromwell.conf?

If this is possible, I cannot see any specific examples in the Cromwell docs. If this is not currently possible, could I request adding the ability to define additional mount points as a feature request??

Kind Regards,
Jon

**Comment by markjschreiber** [comment]
Hi Jon,

This isn't directly supported right now. You could change the
CloudFormation template so that your customized mount is mounted to the EC2
nodes. It is also possible to mount EFS directly into an AWS Batch
container through the job definition but that would require changes in
Cromwell's AWS Batch backend. Using EFS with containers for Cromwell
workflows is something we are investigating but there are some stress tests
that we need to do at scale to see if sufficient IOPs are available.

**Comment by microbioticajon** [comment]
HI @markjschreiber,

Thanks for getting in touch. I had a look at the code and believe I came up with a work around, albeit a bit of a hack. Im building our system in Terraform so have a bit more control over the underlying infrastructure. Im able to mount additional volumes to the underlying ec2 instance, I just needed to mount them inside the container.  I managed this with the following runner config:
    backend {
        default = AWSBatch

        providers {
          AWSBatch {
            actor-factory = "cromwell.backend.impl.aws.AwsBatchBackendLifecycleActorFactory"
           config {
             ...
             default-runtime-attributes {
               queueArn: "${default_batch_queue}"
               scriptBucketName: "${script_bucket}"
               disks: "local-disk,/bin/bash,${static_ref_snapshot_mount}"
               zones: "${aws_batch_availability_zone}"
             }
           }
         }
       }

Specifically `default-runtime-attributes.disks`. The first entry in this list always defaults to the working dir (this is the volume I have set up to autoscale), but after that subsequent values correspond to locations on the host. The only downside is that there is no way to change the mount location in the container. Whatever the location on the host has to be the same location in the container. As I control both I can make sure they are in sync.

It feels a bit clunky but seems to be working. I will close this issue but would suggest some minor modifications to the AWS volume handling could really add to the systems flexibility /:-)

Best,
Jon
